### ISSUE TITLE
Quality & Stutter of instruments

### ISSUE THREAD

**Issue by Uagr** [torilmud/issues#776]
No idea what is good what is bad, is high quality low stutter good? or low quality high stutter? cant find a helpfile anywhere, and no noticeable difference in instrument held for any song.

Examples
mandolin of dark stone and wood
Instrument Type: Mandolin, Quality: 8, Stutter: 8, Min Level: 25
beautiful rosewood mandolin
Instrument Type: Mandolin, Quality: 3, Stutter: 2, Min Level: 0
rosewood mandolin
Instrument Type: Mandolin, Quality: 12, Stutter: 0, Min Level: 35

based on name, I would think that beautiful is best, based on level requirements I think non beautiful is best....If high stutter is needed would think darkwood is... but NFC ...Just guessing I would think you would want high quality low stutter based on the words meaning.... but even Tia drum has high both so really confused, an explination would be appreciated.

You feel informed:
Name 'a small oaken drum wrapped with dragonhide'
Keyword 'drum oaken dragonhide', Item type: INSTRUMENT
Item can be worn on:  HOLD 
Item will give you following abilities:  NOBITS
Item is: GLOW FLOAT NOBURN NOLOCATE NOBITS
Weight: 1, Value: 34534
Instrument Type: Drums, Quality: 12, Stutter: 10, Min Level: 45
Can affect you as :
    Affects : HITPOINTS By 35
    Affects : SV_SPELL By -5

**Comment by vitaladsr** [comment]
Yeah .. good question. Would love to know as well. If I had to pull a complete random guess out of thin air I would think quality had affect on the power of the song and stutter actually meant the higher value the lower the chance of messing up. I have a strange hunch they both don't do anything lol ... But no idea .. prob some orig diku code shite ... Shevy?!

**Comment by Inames** [comment]
> Yeah .. good question. Would love to know as well. If I had to pull a complete random guess out of thin air I would think quality had affect on the power of the song and stutter actually meant the higher value the lower the chance of messing up. I have a strange hunch they both don't do anything lol ... But no idea .. prob some orig diku code shite ... Shevy?!

TL;DR -- Yes this is correct, Quality allows the song to hit peak effectiveness quicker, and stutter is basically a bonus to the stutter save that causes you to expend mana to keep a song going. higher numbers are good on both. Stutter is more important if you don't plan to swap songs much if at all. Quality is great for frequent song switching.

What it means to hit peak effectiveness, is that all songs have a minimum effect and a maximum effect, and no stat, skill, or instrument can boost it beyond the maximum, For example, Heal song ticks for like 90 or 92 health or something MAX per verse, haste/hit/dam song can only get you haste and 10/10 at MAX, but usually songs take a number of verses to ramp up from like 10% to 100% and the speed at which this happens is based on your skill in the song type (offensive, healing, harmful, magical), your skill in the instrument type (if you are using an instrument), and the quality of the instrument. Typically you only need 2-3 verses to get maximum effectiveness with good skills (75+ish) and any appropriate instrument, but with a high quality instrument, 9-12ish you will get maximum effectiveness with decent (60-75ish) skills from round 1.

You can hit peak performance after like 8-10 verses without any instrument if your song skills are high (85+) and don't need instruments, but if you need to swap songs frequently, it's good to use them, like if you are singing haste on Magma and suddenly 4 drakes wander in, you swap to heal, you really want to have an instrument in that case to get to the max tick immediately, rather than having 60hp song ticks for 20 rounds.

Stutter isn't super important because you can get mana back pretty quickly with renewal if you are swapping to it during mem times, and the Virtuoso skill has a significant benefit to stutter checks/mana consumption when you've been singing a song for a while. But if you want to be able to leave haste song running for 2-3 hours uninterrupted, a good stutter instrument will go a long way for that. Higher level songs also have a higher stutter save requirement, even with a max stutter stat on a drum you will likely run out of mana singing elements in about 10 minutes.

Disclaimer - Numbers were approximations to create context, actual values required for described effectiveness may vary. And without looking at the code, much of the mechanics explained are knowledge gained from experience and inference.

**Comment by shevarash** [comment]
To be perfectly honest, I believe this will be resolved with Bard class revamp, which is planned.
